x509util/extensions.go

```diff
@@ -66,6 +66,7 @@ const (
 	RegisteredIDType        = "registeredID"
 	PermanentIdentifierType = "permanentIdentifier"
 	HardwareModuleNameType  = "hardwareModuleName"
+	UPNType                 = "upn"
```

Comment:
Would propose to use `UserPrincipalNameType` internally instead of `UPNType`, but ain't blocking this from being used.

Besides that, maybe it should be `"userPrincipalName"` as the JSON property? That's more in line with the rest of the properties. Curious if it would be possible to use two names for the same, like an alias?

Reply:
Fixed with b724aa2